conan/tools/env/environment.py

```diff
@@ -0,0 +1,138 @@
+import textwrap
```

Comment:
maybe this one deserves its own first class generator? and new Virtual*Env generators might be based on that one?

Reply:
Not yet. This could be a first class model, like ``self.env_info = Environment()`` in the recipes. This is separated in another branch and will have a separate PR once it is stabilized.